### ISSUE TITLE
Error when using subject for action controller

### ISSUE THREAD

**Issue by txl19** [rails-engine/role_core#29]
## Environment
* Ruby version: ruby 3.0.3p157 (RVM to manage)
* Rails version: 7.0.2.3
* Bundler version: 2.2.32
------------------------
## Error description
Using user has single role

- I created role:

```
#<Role:0x00000001067e6ab0
 id: 1,
 name: "Admin",
 permissions: #<Global:OptionsModel {:admin=>{:list_user=>true}}>,
 type: "Role",
 created_at: Fri, 18 Mar 2022 04:14:04.399774000 UTC +00:00,
 updated_at: Fri, 18 Mar 2022 04:14:04.401617000 UTC +00:00>
```

- Define in role_core.rb:

```
require "role_core/contrib/can_can_can_permission"
RoleCore.permission_class = RoleCore::CanCanCanPermission

RoleCore.permission_set_class.draw do
  group :admin, subject: :manage_users do
    permission :list_user, action: :index
  end
end.finalize!
```

- Define in users_controller.rb:

``` 
def index
    authorize! :index, :manage_users

    @users = User.all

    render json: @users
end
```

Error: When i run `authorize! :index, :manage_users`, raise exception `"exception": "#<NoMethodError: undefined method `_namespace' for : manage_users:Symbol>",`
------------------------
Note: I see the difference lately is @options, for previous versions `@options={}` instead of `@options={:_namespace=>[:admin], :_priority=>0, :_callable=> true}`

<img width="498" alt="Screen Shot 2022-03-18 at 11 39 54" src="https://user-images.githubusercontent.com/97243591/158938916-1e3d049f-82cd-4b1d-8859-f51edf28ee07.png">

**Comment by txl19** [comment]
ping @jasl

**Comment by jasl** [comment]
can you show me your permission definitions or give me a sample?

**Comment by jasl** [comment]
and cancancan version? I don't follow its change so newer version may break

**Comment by txl19** [comment]
This is my demo application. You can check how I implemented it in detail
https://github.com/txl19/test-role_core

**Comment by txl19** [comment]
@jasl ping

**Comment by jasl** [comment]
I checked your code

You can't add `subject` to `group` https://github.com/txl19/test-role_core/blob/master/config/initializers/role_core.rb#L14

group is sort of a namespace, it just helps you organize permissions.

you have to transfer it to a permission, e.g.

`permission :manage_users, action: :index, subject: :manage_users`

**Comment by jasl** [comment]
But I don't know why raised error, so I will check

**Comment by txl19** [comment]
@jasl I think due to the recent update of gem cancancan. You can see the change in commit https://github.com/CanCanCommunity/cancancan/commit/0563c055fb9131a5271919c3f08a238e1540347a

**Comment by jasl** [comment]
> @jasl I think due to the recent update of gem cancancan. You can see the change in commit [CanCanCommunity/cancancan@0563c05](https://github.com/CanCanCommunity/cancancan/commit/0563c055fb9131a5271919c3f08a238e1540347a)

this commit are not released yet, so not relate that

**Comment by jasl** [comment]
I cloned your demo app, but you don't implement login action, how did you trigger the error?
and your controller will always failed by `NoMethodError (undefined method `computed_permissions' for nil:NilClass` because of you're missing `load_resource` step (which `cancancan` required)

**Comment by txl19** [comment]
I simulated a session with User.first. You can see  https://github.com/txl19/test-role_core/blob/a8fa6d5c711c94579e9d7fb17a956f4572bf8296/app/controllers/application_controller.rb#L5
This is 'current_ability'
<img width="1262" alt="Screen Shot 2022-03-20 at 11 02 09" src="https://user-images.githubusercontent.com/97243591/159147575-c01bd588-ed3c-4484-9b77-408e6e7572a2.png">
@jasl ping

**Comment by txl19** [comment]
The same definition, with Rails 6 it works fine. I think there's a problem with Rails 7
This is how I do it with Rails 6, you should Run `rails db:seed` to get test data: `https://github.com/txl19/role_core_with_rails6`
User.first has access to `http://localhost:3000/users`
User.second does not have access to `http://localhost:3000/users`

**Comment by txl19** [comment]
@jasl ping

**Comment by jasl** [comment]
This is not an RoleCore bug, it's a Ruby 3.0 bug, the code works great in Ruby 2.7 and 3.1, your Rails 6 demo shows no problem because you're using Ruby 2.7.

The root cause is, in Ruby 3.0.x, the `options` arg of`def initialize(name, _namespace: [], _priority: 0, _callable: true, **options, &block)` includes declared keyword args (`:_namespace, :_priority, :_callable`) which shouldn't be

I make a workaround, released `0.0.28`
